lib/chezmoi/fsmutator_windows.go

```diff
@@ -0,0 +1,148 @@
+// +build windows
+
+package chezmoi
+
+import (
+	"os"
+	"path/filepath"
+	"syscall"
+
+	acl "github.com/hectane/go-acl"
+
+	"github.com/google/renameio"
+	"golang.org/x/sys/windows"
+
+	vfs "github.com/twpayne/go-vfs"
+	"github.com/twpayne/go-vfs/vfst"
+)
+
+func (a *FSMutator) getCorrectedPath(file string) string {
+	var pathfs *vfs.PathFS
+
+	// if it just is a pathfs, cool
+	maybe_pathfs, ok := a.FS.(*vfs.PathFS)
+	if !ok {
+		maybe_testfs, ok := a.FS.(*vfst.TestFS)
+
+		if ok {
+			pathfs = &maybe_testfs.PathFS
+		}
+	} else {
+		pathfs = maybe_pathfs
+	}
+
+	if pathfs != nil {
+		joined, err := pathfs.Join("", file)
+		if err == nil {
+			file = joined
+		}
+	}
+
+	return file
+}
+
+// Use the same default value as Linux (as of kernel 4.19)
+const MAXSYMLINKS = 40
+
+func (a *FSMutator) IsPrivate(file string, umask os.FileMode) bool {
+	file = a.getCorrectedPath(file)
+
+	// if file is a symlink, get the path it links to.  this emulates
+	// unix-style behavior, where symlinks can't have their own independent
+	// permissions.
+
+	resolved := file
+	for i := 0; i < MAXSYMLINKS; i++ {
+		fi, err := os.Lstat(resolved)
+		if err != nil {
+			return false
+		}
+
+		if fi.Mode()&os.ModeSymlink == 0 {
+			// not a link, all done
+			break
+		}
+
+		next, err := os.Readlink(resolved)
+		if err != nil {
+			return false
+		}
+
+		if next != "" && !filepath.IsAbs(next) {
+			resolved = filepath.Join(filepath.Dir(resolved), next)
+		}
+	}
+
+	mode, err := acl.GetEffectiveAccessMode(resolved)
+	if err != nil {
+		return false
+	}
+
+	return (uint32(mode) & 0007) == 0
+}
+
+// WriteFile implements Mutator.WriteFile.
+func (a *FSMutator) WriteFile(name string, data []byte, perm os.FileMode, currData []byte) error {
+	// Special case: if writing to the real filesystem, use github.com/google/renameio
```

Comment:
After a private chat with @stapelberg last week (author of `github.com/google/renameio`) I understood that atomic file replacement is not possible on Windows, so we can remove the use of `renameio` here. I've [asked for confirmation](https://github.com/google/renameio/issues/1#issuecomment-508942124).

Reply:
I see.  I've left it in for now but once he confirms I'll go ahead and remove it.